web-services/common/src/main/java/datawave/webservice/common/cache/SharedCacheCoordinator.java

```diff
@@ -332,19 +333,34 @@ public InterProcessLock getMutex(String path) {
      *            a listener that is called when the counter value changes
      */
     public void registerCounter(String counterName, SharedCountListener listener) throws Exception {
+        Preconditions.checkArgument(!sharedCounters.containsKey(counterName), "Counter " + counterName + " has already been registered!");
         reregisterCounter(counterName, listener, 1);
     }
     
     private void reregisterCounter(String counterName, SharedCountListener listener, int seedValue) throws Exception {
         ArgumentChecker.notNull(counterName, listener);
-        Preconditions.checkArgument(!sharedCounters.containsKey(counterName), "Counter " + counterName + " has already been registered!");
         
         SharedCount count = new SharedCount(curatorClient, ZKPaths.makePath("/counters", counterName), seedValue);
         count.start();
         sharedCounters.put(counterName, count);
         sharedCountListeners.put(counterName, listener);
         localCounters.put(counterName, count.getCount());
         
+        count.addListener(new SharedCountListener() {
```

Comment:
I was seeing a failure in the SharedCacheCoordinator test after updating the curator version. I believe we were incorrectly using the API. When a shared object (counter, boolean, etc) loses the connection to zookeeper, the listeners must be re-registered upon re-connection.